packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1787,6 +1837,120 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState(null);
+  let [ascDesc, setAscDesc] = useState('desc');
+
+  function onSortTable(e) {
+    let transform =
+      e.currentTarget.innerText.toLowerCase() === 'payment' ||
+      e.currentTarget.innerText.toLowerCase() === 'deposit'
+        ? 'amount'
+        : e.currentTarget.innerText.toLowerCase();
+    if (transform === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(transform);
+      transform === 'amount' || transform === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = [field] + 's';
+
+  function sortStrings() {
+    transactions =
+      ascDesc === 'asc'
+        ? transactions
+            .map(e => {
+              return {
+                ...e,
+                test: props[fields].find(f => {
+                  return f.id === e[field];
+                })
+                  ? props[fields].find(f => {
+                      return f.id === e[field];
+                    }).name
+                  : '',
+              };
+            })
+            .sort((a, b) => a.test < b.test)
+        : transactions
+            .map(e => {
+              return {
+                ...e,
+                test: props[fields].find(f => {
+                  return f.id === e[field];
+                })
+                  ? props[fields].find(f => {
+                      return f.id === e[field];
+                    }).name
+                  : '',
+              };
+            })
+            .sort((a, b) => a.test > b.test);
+  }
+
+  function sortCat() {
+    transactions =
+      ascDesc === 'asc'
+        ? transactions
+            .map(e => {
+              return {
+                ...e,
+                test: category.find(f => {
+                  return f.id === e[field];
+                })
+                  ? category.find(f => {
+                      return f.id === e[field];
+                    }).name
+                  : '',
+              };
+            })
+            .sort((a, b) => a.test < b.test)
+        : transactions
+            .map(e => {
+              return {
+                ...e,
+                test: category.find(f => {
+                  return f.id === e[field];
+                })
+                  ? category.find(f => {
+                      return f.id === e[field];
+                    }).name
+                  : '',
+              };
+            })
+            .sort((a, b) => a.test > b.test);
+  }
+
+  if (field) {
+    switch (field) {
```

Comment:
I think the `switch` will hit the `default` case if `field` is not set, right? (Alternatively, you could start `field` off as `'date'` so that there is always a valid sort field).